crates/bevy_ecs/src/query/filter.rs

```diff
@@ -938,6 +942,11 @@ impl<T: Component> QueryFilter for Changed<T> {
 ///
 /// [`Added`] and [`Changed`] works with entities, and therefore are not archetypal. As such
 /// they do not implement [`ArchetypeFilter`].
+#[diagnostic::on_unimplemented(
+    message = "`{Self}` is not a valid `Query` filter based on archetype information",
+    label = "invalid `Query` filter",
+    note = "an `ArchetypeFilter` typically uses a combination of `With<T>` and `Without<T>` statements"
```

Comment:
(Nit) Would it be worth mentioning that `ArchetypeFilter` _does not_ support `Added` and `Changed`? Or at least only a subset of `QueryFilter`s are `ArchetypeFilter`s? I worry that someone would confuse the two traits.

Reply:
If we _knew_ that the issue was caused by using one of those types I'd definitely want the info included. But since we don't I'd worry about overloading the user with information that might not help.

Reply:
Ok, sounds good!